ArcGISRuntimeSDKQt_CppSamples/Layers/QueryMapImageSublayer/QueryMapImageSublayer.cpp

```diff
@@ -193,7 +170,19 @@ void QueryMapImageSublayer::query(const QString& whereClause)
   queryParams.setWhereClause(whereClause);
 
   // query the feature tables
-  m_citiesTable->queryFeatures(queryParams);
-  m_countiesTable->queryFeatures(queryParams);
-  m_statesTable->queryFeatures(queryParams);
+  m_citiesTable->queryFeaturesAsync(queryParams).then(this, [this](FeatureQueryResult* result)
+  {
+    addResultsAsGraphics(result, m_citySymbol);
+    delete result;
+  });
+  m_countiesTable->queryFeaturesAsync(queryParams).then(this, [this](FeatureQueryResult* result)
+  {
+    addResultsAsGraphics(result, m_countySymbol);
+    delete result;
+  });
+  m_statesTable->queryFeaturesAsync(queryParams).then(this, [this](FeatureQueryResult* result)
+  {
+    addResultsAsGraphics(result, m_stateSymbol);
+    delete result;
```

Comment:
Again, I realize this is not your code, but please do some minor refactoring to remove any `delete` calls. We should use newer RAII techniques whenever possible.

Reply:
Replaced raw pointers with `unique_ptr` wherever `delete` is used.

Reply:
Reverted the changes made.